src/path/index.ts

```diff
@@ -62,9 +62,12 @@ export function root(): DirectoryPath {
 /**
  * Utility function create an app data path.
  */
-export function appData(app: AppInfo, suffix?: DistinctivePath): DistinctivePath {
+export function appData(app: AppInfo): DirectoryPath
+export function appData(app: AppInfo, suffix: FilePath): FilePath
+export function appData(app: AppInfo, suffix: DirectoryPath): DirectoryPath
+export function appData(app: AppInfo, suffix: DistinctivePath): DistinctivePath
+export function appData(app: AppInfo, suffix?: any): any {
```

Comment:
Could this be written without using `any`? 

It might work to replace the `any` with `DistinctivePath` because it is broad enough to cover each of the `suffix` overloads and the return type overloads.

Reply:
Done 👍 I didn't realise you could have two times the same function type with overloads, because the last type signature isn't exported and is only used for the implementation.